python/lsst/cp/pipe/linearity.py

```diff
@@ -192,75 +232,237 @@ def run(self, inputPtc, camera=None, inputDims=None):
                     timeRange = np.linspace(0, tMax, self.config.maxLookupTableAdu)
                     signalIdeal = polyFit[0] + polyFit[1]*timeRange
                     signalUncorrected = funcPolynomial(polyFit, timeRange)
-                    lookupTableRow = signalIdeal - signalUncorrected  # LinearizerLookupTable has corrections
+                    lookupTableRow = signalIdeal - signalUncorrected  # LinearizerLookupTable has correction
 
                     linearizer.tableData[tableIndex, :] = lookupTableRow
                     linearityFit = [tableIndex, 0]
                     tableIndex += 1
+            elif self.config.linearityType in ['Spline']:
+                # See discussion in `lsst.ip.isr.linearize.py` before modifying.
+                numPerBin, binEdges = np.histogram(linearOrdinate, bins=self.config.fitOrder)
+                with np.errstate(invalid="ignore"):
+                    values = np.histogram(linearOrdinate, bins=self.config.fitOrder,
+                                          weights=(inputAbscissa[fluxMask] - linearOrdinate))[0]/numPerBin
+                    binCenters = np.histogram(linearOrdinate, bins=self.config.fitOrder,
+                                              weights=linearOrdinate)[0]/numPerBin
```

Comment:
Ignoring the variable name issue mentioned above, it might be easiest to show by example.  If we take the histogram of [1, 2, 3, 4, 5, 6] with `fitOrder = 3`, then `numPerBin = [2, 2, 2]`.  The binCenters histogram values are then [1+2, 3+4, 5+6], which divided by the number per bin gives the expected [1.5, 3.5, 5.5] centers.  This is admittedly being tricky instead of being clear, but is an adaptation of how splines are handled for the overscan correction.